### PR TITLE
chore(schematics): add simple-import-sort & unused-imports rules

### LINKED ISSUE #1548: chore: cleanup eslint rules
# Description

We have plenty of eslint activated that make no sense for our library and get disabled manually all over the place.

* @angular-eslint/directive-selector
* @angular-eslint/directive-class-suffix
* @angular-eslint/component-selector
* @angular-eslint/component-class-suffix
* @angular-eslint/no-input-rename
* @typescript-eslint/member-ordering
* @typescript-eslint/no-non-null-assertion -> I would rather suggest finally going for strict mode

On the other side we are missing some very useful eslint rules:

* [simple-import-sort](https://www.npmjs.com/package/eslint-plugin-simple-import-sort)/imports
* [simple-import-sort](https://www.npmjs.com/package/eslint-plugin-simple-import-sort)/exports
* [unused-imports](https://www.npmjs.com/package/eslint-plugin-unused-imports)

# Tasks

* [ ] clean up eslint rules and unify them. This may also affect the imported plugins
* [ ] remove `// eslint-disable-next-line` comments where possible
* [ ] add new rules as suggested
* [ ] run `eslint --fix` for all files

### PR DESCRIPTION
**Pull request by Naaajii**
this commit installs simple-import-sort & unused-imports eslint rules into workspace

fixes #1548 - add new rules as suggested

### CODE REVIEW

**Comment by hoebbelsB** [comment]
hey @Naaajii , big thanks for your contribution :heart: 

> I have added new rules to the workspace repo, please correct me if I have done anything wrong.

Looks good, let's just configure them as `error`, not as `warn` and run `eslint --fix` afterwards.

> should spec files be included in ignorePatterns?

Let's keep it as is for now

> what should we do with following warnings as they are all over the place:
Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
Empty block statement                     no-empty
'intersectionObserver' is defined but never used  @typescript-eslint/no-unused-vars
remove them? disable them ( tho doing that would be same as doing what we are supposed to in this issue )? fix them one-by-one? I'm sure we can't just fix them by passing --fix as we can for unused imports or sorted imports/exports.

Ignore the warnings for now, we run the lint with `--quiet` to only show errors in the ci pipeline.

> additionally - [eslint-plugin-unused-imports](https://github.com/sweepline/eslint-plugin-unused-imports) comes with no-unused-vars which is same as [@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars). I am unsure what to keep.

Let's only go with one of them. I would suggest to keep `@typescript-eslint/no-unused-vars` in place

File: .eslintrc.json [hunk 1]
```diff
@@ -30,7 +30,11 @@
               }
             ]
           }
-        ]
+        ],
+        "simple-import-sort/imports": "warn",
+        "simple-import-sort/exports": "warn",
+        "no-unused-vars": "off",
+        "unused-imports/no-unused-imports": "warn"
```
Comment (hoebbelsB) [inline .eslintrc.json:37]: let's configure them as `error` and run `nx affected:lint --fix` afterwards :)

**Review by hoebbelsB — CHANGES_REQUESTED** [review]

**Review by edbzn — APPROVED** [review]

**Comment by edbzn** [comment]
@Naaajii Can this PR be considered as ready for merge?

**Comment by Naaajii** [comment]
@edbzn I believe I did implement most of what issue initially mentioned, so yes. however if something looks off let me know after review.

**Comment by Naaajii** [comment]
i have removed the conflicts so the PR can move forward.

**Comment by Naaajii** [comment]
aaaaaa, I just rebase and force pushed but conflicts still exists? I might need some help off this. @edbzn

**Pull request closed without merging**